sdk/eidolon_ai_sdk/cpu/conversational_agent_cpu.py

```diff
@@ -140,23 +135,29 @@ async def _call_tool(
         tool_defs,
         conversation: List[LLMMessage],
     ):
-        tool_def = tool_defs[tool_call_event.tool_call.name]
         tc = tool_call_event.tool_call
-        tool_stream = stream_manager(
-            tool_def.execute(tool_call=tc),
-            ToolCallStartEvent(tool_call=tc, context_id=tc.tool_call_id),
-        )
+        logic_unit_wrapper = ["NaN"]
+
+        def tool_event_stream():
+            try:
+                tool_def = tool_defs[tc.name]
+                logic_unit_wrapper[0] = tool_def.logic_unit.__class__.__name__
+                return tool_def.execute(tool_call=tc)
+            except KeyError:
+                raise ValueError(f"Tool {tool_call_event.tool_call.name} not found. Available tools: {tool_defs.keys()}")
+
+        tool_stream = stream_manager(tool_event_stream, ToolCallStartEvent(tool_call=tc, context_id=tc.tool_call_id))
```

Comment:
There was a bug where the llm could call a tool which did not exist, causing us the throw before our context was created (when it should be an error within the context